File: example/src/main/java/autodagger/example/ThirdActivity.java
package autodagger.example;

import android.app.Activity;
import android.os.Bundle;

import autodagger.AutoInjector;

/**
 * Showcase: @AutoComponent from annotation
 *
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
@Deprecated
@StandardActivityComponent1
@AutoInjector
@DaggerScope(ThirdActivity.class)
public class ThirdActivity extends Activity {

    private ThirdActivityComponent component;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        component = DaggerThirdActivityComponent.builder()
                .myAppComponent(((MyApp) getApplication()).getComponent())
                .build();
        component.inject(this);
    }
}

File: example/src/main/kotlin/autodagger/example/second/ThirdActivity.kt
package autodagger.example.second

import android.app.Activity
import android.os.Bundle
import autodagger.AutoInjector
import autodagger.example.DaggerScope
import autodagger.example.KotlinExampleApplication

/**
 * Showcase: @AutoComponent from annotation
 */
@StandardActivityComponent1
@AutoInjector
@DaggerScope(ThirdActivity::class)
class ThirdActivity : Activity() {

    private val component: ThirdActivityComponent by lazy {
        DaggerThirdActivityComponent.builder()
                .kotlinExampleApplicationComponent((application as KotlinExampleApplication).component)
                .build()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        component.inject(this)
    }
}
